wallet-api/src/Wallet/Emulator/Types.hs

```diff
@@ -418,10 +418,10 @@ processPending = Op.singleton BlockchainProcessPending
 addBlocks :: Int -> Trace m [Block]
 addBlocks i = traverse (const processPending) [1..i]
 
+-- | Add a number of blocks, notifying all wallets after each block.
 addBlocksAndNotify :: [Wallet] -> Int -> Trace m ()
-addBlocksAndNotify wallets i = do
-  blocks <- addBlocks i
-  traverse_ (\_ -> processPending >>= walletsNotifyBlock wallets) blocks
+addBlocksAndNotify wallets i = 
+    traverse_ (processPending >>= walletsNotifyBlock wallets) [1..i]
```

Comment:
This seems to be missing the actual adding of the block?

Reply:
Adding the block happens in `processPending` (that's what `addBlocks` does: `addBlocks i = traverse (const processPending) [1..i])` So the previous code was wrong in that it actually added twice the number of blocks.

Reply:
ahhh, I had the same thought when you suggested that I could change this before, sorry I should have queried you then.

Reply:
Ah right of course.